x-pack/plugins/observability_ai_assistant/server/service/client/adapters/bedrock_claude_adapter.ts

```diff
@@ -0,0 +1,224 @@
+      even if the "recall" function was executed after that. Consider the tools you need to answer the user's question.
+      
+      Rather than explaining how you would call a function, just generate the XML to call the function. It will automatically be
+      executed and returned to you.
+
+      These results are generally not visible to the user. Treat them as if they are not,
+      unless specified otherwise.
+
+      ONLY respond with XML, do not add any text.
+
+      If a parameter allows multiple values, separate the values by ","
+      
+      You may call them like this.
+
+      <function_calls>
+        <invoke>
+          <tool_name>$TOOL_NAME</tool_name>
+          <parameters>
+            <$PARAMETER_NAME>$PARAMETER_VALUE</$PARAMETER_NAME>
+            ...
+          </parameters>
+        </invoke>
+      </function_calls>
+
+      Here are the tools available:
+
+      <tools>
+        ${filteredFunctions
+          .map(
+            (fn) => `<tool_description>
+          <tool_name>${fn.name}</tool_name>
+          <description>${fn.description}</description>
+          <parameters>
+            ${jsonSchemaToFlatParameters(fn.parameters).map((param) => {
+              return `<parameter>
+                <name>${param.name}</name>
+                <type>${param.type}</type>
+                <description>
+                  ${param.description || ''}
+                  Required: ${!!param.required}
+                  Multiple: ${!!param.array}
+                  ${
+                    param.enum || param.constant
+                      ? `Allowed values: ${castArray(param.constant || param.enum).join(', ')}`
+                      : ''
+                  }
+                </description>
+              </parameter>`;
+            })}
+          </parameters>
+        </tool_description>`
+          )
+          .join('\n')}
+      </tools>
+      
+      
+      Examples:
+
+      Assistant:
+      <function_calls>
+        <invoke>
+          <tool_name>my_tool</tool_name>
+          <parameters>
+            <myParam>foo</myParam>
+          </parameters>
+        </invoke>
+      </function_calls>
+
+      Assistant: 
+      <function_calls>
+        <invoke>
+          <tool_name>another_tool</tool_name>
+          <parameters>
+            <myParam>foo</myParam>
+          </parameters>
+        </invoke>
+      </function_calls>
+
+      `;
+    }
+
+    const formattedMessages = [
+      {
+        role: 'system',
+        content: `${replaceFunctionsWithTools(systemMessage.message.content!)}
+          
+          ${functionsPrompt}
+          `,
+      },
+      ...otherMessages.map((message, index) => {
+        const builder = new Builder({ headless: true });
+        if (message.message.name) {
+          const deserialized = JSON.parse(message.message.content || '{}');
```

Comment:
Do we want to use `x-pack/plugins/observability_ai_assistant/public/utils/safe_json_parse.ts` here?

Reply:
I'm not 100% sure, but I'm leaning towards having this explicitly fail and return an error from the API call. In the UI it is slightly different, as you cannot see a conversation anymore if there is invalid JSON and we don't use `safeJsonParse`. (In this case, it actually made a bug more visible).